backend/ml/tests/test_mehestan_primitives.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+from django.test import TestCase
+
+from ml.mehestan.primitives import QrMed, QrUnc
+
+
+class QrMedTest(TestCase):
+    def test_qrmed(self):
+        W = 2
+        weight = 1
+        self.assertAlmostEqual(
+            QrMed(
+                W,
+                w=weight,
+                x=np.array([-10.0, 1.0, 10.0]),
+                delta=np.array([1e-3, 1e-3, 1e-3]),
+            ),
+            0.5, # influence of the median contributor is bounded by weight / W
+            places=3
+        )
+
+    def test_qrmed_with_W_equals_zero(self):
+        W = 0
+        weight = 1
+        self.assertAlmostEqual(
+            QrMed(
+                W,
+                w=weight,
+                x=np.array([-10.0, 1.0, 10.0]),
+                delta=np.array([1e-3, 1e-3, 1e-3]),
+            ),
+            1.0,
+            places=3
+        )
+
+
+class QrUncTest(TestCase):
```

Comment:
Are these tests created by running the algorithm first and then writing the test with the output value? If not, could we have comments that explain how the values are obtained, like you did for `QrMedTest.test_qrmed`?

I wish we would somehow be able to see the properties of QrUnc in the test suite (e.g. "always positive" or "always between zero and the true median" tested with multiple random inputs).

I say this while barely understanding QrUnc. Happy to be taught everything.